### PR TITLE
[XCBuildSupport] Add PlatformFilter to build file and target dependency

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
This will be used for implementing target conditionals without
requiring PIF regeneration.

<rdar://problem/59535847>

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

File: Sources/XCBuildSupport/PIF.swift [hunk 1]
```diff
@@ -550,6 +576,24 @@ public enum PIF {
...
+    /// Represents a generic platform filter.
+    public struct PlatformFilter: Encodable {
+        /// The name of the platform (`LC_BUILD_VERSION`).
+        ///
+        /// Example: macos, ios, watchos, tvos.
+        public var platform: String
+
+        /// The name of the environment (`LC_BUILD_VERSION`)
+        ///
+        /// Example: simulator, macabi.
```
Comment (jakepetroules) [inline Sources/XCBuildSupport/PIF.swift]: This should be `maccatalyst` for Mac Catalyst, not `macabi`.

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test self hosted

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Pull request merged**